samples/testing-frameworks/appium/client-side/csharp/biometric-sample-test/BitbarBiometricSample/TestBiometricsIOS.cs

```diff
@@ -0,0 +1,83 @@
+﻿using OpenQA.Selenium.Appium.iOS;
+using OpenQA.Selenium.Support.UI;
+
+namespace BitbarBiometricSample
+{
+    [TestFixture]
+    public class TestBiometricsIOS
+    {
+        private readonly String SCREENSHOT_DIRECTORY = AppDomain.CurrentDomain.BaseDirectory + "/screenshots/";
+        private RemoteWebDriver driver;
+
+        [OneTimeSetUp]
+        public void InitDriver()
+        {
+            const String AppiumHubURL = "https://appium.bitbar.com/wd/hub";
+
+            AppiumOptions capabilities = new();
+            //Platfrom dependent 
+            capabilities.AddAdditionalCapability("platformName", "iOS");
+            capabilities.AddAdditionalCapability("deviceName", "iPhone device");
+            capabilities.AddAdditionalCapability("automationName", "XCUITest");
+            //Customizable
+            capabilities.AddAdditionalCapability("bitbar_project", "C# Appium Automated Test");
+            capabilities.AddAdditionalCapability("bitbar_testrun", "iOS Run");
+            capabilities.AddAdditionalCapability("bitbar_device", "Apple iPhone 11");
+            //Required
+            capabilities.AddAdditionalCapability("bitbar_apiKey", "<YOUR_BITBAR_API_KEY>");
+            capabilities.AddAdditionalCapability("bitbar_app", "<APP_ID>");
+            //Biometrics capability
+            capabilities.AddAdditionalCapability("biometricInstrumentation", true);
+
+            driver = new IOSDriver<IOSElement>(new Uri(AppiumHubURL), capabilities, TimeSpan.FromSeconds(300));
+        }
+
+        [OneTimeTearDown]
+        public void QuitDriver()
+        {
+            driver.Quit();
+        }
+
+        [Test]
+        public void Test_Biometric_Authentication()
+        {
+            Console.WriteLine("Test execution started:");
+
+            //Accept initial biometric dialog
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10)).Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//XCUIElementTypeAlert[@name=\"Biometric Authentication\"]")));
+            driver.ExecuteScript("mobile: alert", new Dictionary<string, string> { { "action", "accept" }, { "buttonLabel", "OK" } });
+
+            TakeScreenshot("HomeScreen");
+            var buttonBiometrics = driver.FindElement(By.XPath("(//*[@name=\"Biometric authentication\"])[3]"));
+            buttonBiometrics.Click();
+
+            var textSensorType = driver.FindElement(MobileBy.AccessibilityId("Sensor type value"));
+            Assert.That(textSensorType.Text.ToUpper, Is.EqualTo("TOUCHID"));
```

Comment:
I'm not sure if checking sensor type is a good idea. That test may be launch on iPhones with FaceID as well. If you really want to check sensor type, I think checking if it's not `NOT AVAILABLE` seems to be better idea. Alternatively check if it is `TOUCHID` or `FACEID`.

Reply:
Good idea.